service-base/src/main/java/org/eclipse/hono/service/registration/RegistrationHttpEndpoint.java

```diff
@@ -42,7 +44,8 @@
  */
 public final class RegistrationHttpEndpoint extends AbstractHttpEndpoint<ServiceConfigProperties> {
 
-    private static final String PARAM_TENANT = "tenant";
+    // path parameters for capturing parts of the URI path
+    private static final String PARAM_TENANT_ID = "tenant_id";
     private static final String PARAM_DEVICE_ID = "device_id";
```

Comment:
IMHO we should do
````
String PARAM_TENANT_ID =  RequestResponseApiConstants.FIELD_TENANT_ID;
String PARAM_DEVICE_ID = RequestResponseApiConstants.FIELD_DEVICE_ID;
````
we then do not need to introduce the additional comparison in the `getPayloadForParams()` method and the code will be less confusing

Reply:
I agree that this would be less confusing. But using a dash (as in "device-id") in a path parameter doesn't seem to be supported by vertx-web. The parameters can't be referenced that way.

Reply:
I see. Too bad ...